### PR TITLE
Donation form

### PR DESCRIPTION
**Pull request by mikeyrayvon**
- Relies on [odie-donate](https://github.com/interglobalvision/odie-donate) for charge processing
- `yarn install` for [react-stripe-elements](https://github.com/stripe/react-stripe-elements)
- [card details test values](https://stripe.com/docs/testing#cards)

### CODE REVIEW

File: src/components/Donate.jsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
+// MyStoreCheckout.js
+import React, { Component } from 'react';
+import AsciiOdie from './AsciiOdie'
+import Loading from './Loading'
+import DonateForm from './DonateForm'
+import {StripeProvider} from 'react-stripe-elements';
+import {Elements} from 'react-stripe-elements';
+import {StripeKeys} from '../utilities/stripeKeys';
+
+class Donate extends Component<{}, {stripe: null | StripeShape}> {
```
Comment (varffs) [inline src/components/Donate.jsx]: What does `{}, {stripe: null | StripeShape}> {` do?

File: src/components/Donate.jsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    this.stripeKey = StripeKeys.test;
+
+    this.state = {
+      stripe: null,
+    };
+  }
+
+  componentDidMount() {
+    // inject Stripe.js
```
Comment (varffs) [inline src/components/Donate.jsx:22]: Is this ever removed from the DOM?

File: src/components/DonateForm.jsx [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+          </label>
+
+          <div id="form-odie-description" className='grid-item item-s-12 margin-bottom-small'>
+            <textarea
+              id='note'
+              name='note'
+              placeholder='note'
+              disabled={this.state.isLoading}
+              value={this.state.note}
+              onChange={ event => this.setState({ note: event.target.value })}
```
Comment (varffs) [inline src/components/DonateForm.jsx:151]: this is unsanitized text which might be something we have to check?

**Review by varffs — COMMENTED** [review]
Functions correctly. A few questions.

In terms of UX/UI I think the success notification is too minimal. Might even need to remove the form or something and just say thanks.

& the outline on the slider kinda clashes with the styling: 
screen shot 2018-06-15 at 22 14 25

File: src/components/Donate.jsx (hunk 2 above)
Comment (mikeyrayvon) [reply src/components/Donate.jsx:22]: Stripe recommends keeping it on all pages. We might want to inject it even earlier.

File: src/components/Donate.jsx (hunk 1 above)
Comment (mikeyrayvon) [reply src/components/Donate.jsx]: No clue. that is from their example code for injecting stripe.js

Comment (mixcas) [reply src/components/Donate.jsx]: so cryptic

File: src/components/DonateForm.jsx (hunk 3 above)
Comment (mikeyrayvon) [reply src/components/DonateForm.jsx:151]: Yes we should

Comment (mikeyrayvon) [reply src/components/DonateForm.jsx:151]: https://stackoverflow.com/questions/6659351/removing-all-script-tags-from-html-with-js-regular-expression

**Comment by mikeyrayvon** [comment]
I agree about the :focus outlines. I'll make a trello task

**Pull request merged**